recipe/deploy/update_code.php

```diff
@@ -114,6 +115,13 @@
         run("$git clone -l $bare .");
         run("$git remote set-url origin $repository", env: $env);
         run("$git checkout --force $target");
+    } elseif (get('update_code_strategy') === 'clone_submodule') {
+        cd('{{release_path}}');
+        run("$git clone -l $bare .");
+        run("$git remote set-url origin $repository", env: $env);
+        run("$git checkout --force $target");
+        run("$git submodule init");
```

Comment:
Maybe it's better to define it as a separate task which can be attached if needed.

Reply:
I'm sorry, I'm not much into Deployer. I don't know how to solve this differently. Also, a separate task just for basically one line would be more complex.

Reply:
```php
task('deploy:git:submodules', function () {
    run('git submodule init');
    run('git submodule update');
});
```

Reply:
```php

host('production')
    ->set('needs_submodules', true);

after('deploy:update_code', function () {
    if (get('needs_submodules', false)) {
        invoke('deploy:git:submodules');
    }
});
```